src/engine/source/hlp2/src/parse_field.cpp

```diff
@@ -90,30 +90,34 @@ void updateDoc(json::Json& doc,
                std::string_view hdr,
                std::string_view val,
                bool is_escaped,
-               std::string_view escape)
+               std::string_view escape,
+               bool is_quoted)
 {
     if (val.empty())
     {
         doc.setNull(hdr);
         return;
     }
 
-    int64_t i;
-    auto [ptr, ec] {utils::from_chars(val.data(), val.data() + val.size(), i)};
-    if (std::errc() == ec && (val.data() + val.size()) == ptr)
+    if(!is_quoted)
     {
-        doc.setInt64(i, hdr);
-        return;
-    }
-    else
-    {
-        double_t d;
-        auto [ptr, ec] {utils::from_chars(val.data(), val.data() + val.size(), d)};
+        int64_t i;
+        auto [ptr, ec] {utils::from_chars(val.data(), val.data() + val.size(), i)};
         if (std::errc() == ec && (val.data() + val.size()) == ptr)
         {
-            doc.setDouble(d, hdr);
+            doc.setInt64(i, hdr);
             return;
         }
+        else
+        {
+            double_t d;
+            auto [ptr, ec] {utils::from_chars(val.data(), val.data() + val.size(), d)};
+            if (std::errc() == ec && (val.data() + val.size()) == ptr)
+            {
+                doc.setDouble(d, hdr);
+                return;
+            }
+        }
     }
 
     auto vs = std::string {val.data(), val.size()};
```

Comment:
A version with these bugs fixed would be something like:
```cpp

void updateDoc(json::Json& doc,
               std::string_view header,
               std::string_view value,
               bool is_escaped,
               std::string_view escape,
               bool is_quoted)
{
    if (value.empty())
    {
        doc.setNull(header);
        return;
    }

    // If the value is not quoted, try parsing it as an int or double
    if (!is_quoted)
    {
        // Try parsing as int
        int64_t i;
        auto [ptr, ec] {utils::from_chars(value.data(), value.data() + value.size(), i)};
        if (std::errc() == ec && (value.data() + value.size()) == ptr)
        {
            doc.setInt64(i, header);
            return;
        }

        // Try parsing as double
        double_t d;
        auto [ptr2, ec2] {utils::from_chars(value.data(), value.data() + value.size(), d)};
        if (std::errc() == ec2 && (value.data() + value.size()) == ptr2)
        {
            doc.setDouble(d, header);
            return;
        }
    }

    // If the value is a string, unescape it if necessary and add it to the JSON document
    auto vs = std::string {val.data(), val.size()};
    unescape(is_escaped, vs, escape);
    doc.setString(vs, header);
}
```
I have not tested it (nor compiled it :rofl: ), but I recommend to take it into account for changes.

Reply:
Ok, I made the changes and tested it, and it works